### PR TITLE
Task 59 : add SeleneCollection_Enumerable_ShouldReturnFirst as example

### PR DESCRIPTION
**Pull request by OlegZarevych**

### CODE REVIEW

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {
+
+        [TearDown]
+        public void TeardownTest()
+        {
+            Configuration.Timeout = 4;
+        }
+
+        [Test]
+        public void SeleneCollection_Enumerable_ShouldReturnFirst()
```
Comment (yashaka) [inline NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:20]: SeleneCollection_Enumerable_Should.SeleneCollection_Enumerable_ShouldReturnFirst

the full name have duplication... let's remove SeleneCollection_Enumerable_Should from the test method name

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            SeleneCollection elements = SS("#will-exist>input");
+
+            When.WithBody(@"
+                <p id='will-exist'>
+                    <input id='ask' type='submit' value='How r u?' style='display:none'></input>
+                    <input id='answer1' type='submit' value='Good!'></input>
+                    <input id='answer2' type='submit' value='Great!'></input>
+                </p>"
+            );
+            var element = elements.ActualWebElements.First();
```
Comment (yashaka) [inline NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:32]: if we check here that SeleneCollection should behave like Enumerable, we should call .First() on elements, not elements.ActualWebElements ...

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            Given.OpenedEmptyPage();
+            SeleneCollection elements = SS("#will-exist>input");
+
+            When.WithBody(@"
+                <p id='will-exist'>
+                    <input id='ask' type='submit' value='How r u?' style='display:none'></input>
+                    <input id='answer1' type='submit' value='Good!'></input>
+                    <input id='answer2' type='submit' value='Great!'></input>
+                </p>"
+            );
```
Comment (yashaka) [inline NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:31]: probably no need to break code into OpenedEmptyPage and then When.WithBody
there should be something like:

Given.PageWithBody(....) that does 2 things at once

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs (hunk 2 above)
Comment (yashaka) [reply NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:32]: though, maybe we have to call it like: ((Enumerable)elements).First() ...
need to check:)

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            var element = elements.ActualWebElements.First();
+
+            Assert.IsFalse(element.Displayed);
+            Assert.AreEqual("input", element.TagName);
+            Assert.IsEmpty(element.Text);
+            Assert.AreEqual("ask", element.GetAttribute("id"));
```
Comment (yashaka) [inline NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:37]: actually... here we just need to check that elements.First() will return the first element among list...
no need to have 4 asserts
something like

element.Should(Have.text("a")) should be enough
where element is the first one among:
```
<ul>
<li class="item">a</li>
<li class="item">b</li>
<li class="item">c</li>
</ul>
```

**Review by yashaka — CHANGES_REQUESTED** [review]

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs (hunk 3 above)
Comment (OlegZarevych) [reply NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:31]: Done

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs (hunk 1 above)
Comment (OlegZarevych) [reply NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:20]: Done

File: NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs (hunk 4 above)
Comment (OlegZarevych) [reply NSeleneTests/Integration/SharedDriver/SeleneCollection_Enumerable_Should.cs:37]: Done

**Pull request merged**